### ISSUE TITLE
Issue about whether and how the spatial resolution would affect the HMM results and its reliability

### ISSUE THREAD

**Issue by lsyoath** [OHBA-analysis/HMM-MAR#46]
Hello,

We've got HMM states based on the network parcellation, and the results confused me. 

Our spatial resolution is 7, so that theoretically we should set the K value between 2 and 6 in my own understanding. And actually when we set K value 6, we only got 5 states. I'm not sure whether I've done something wrong, or it is constrained by the spatial resolution. 

Another question is about the probability matrix, cause it showed nothing meaningful in our results, especially when K = 2, 3 and 4. I guess that may be on account of the too small number of states, but I’m not sure. Can you give me some directions?

Thank you so much for your kindness and help.

Best,
Siya

**Comment by vidaurre** [comment]
Hi, 

I'm not completely sure what you mean by the spatial resolution in this context, and how that relates to the choice about the number of states. In any case, it can happen that some states get eliminated during training, if the inference process considers they are not necessary. With the options.dropstates = 0 , the unnecessary states should be deactivated but not eliminated.

About the transition probability matrix, can you further specify what you mean by "not meaningful" ?

Thanks

**Comment by lsyoath** [comment]
Hi,
Thanks for replying. I'm sorry for my confusing descriptions.

'Spatial resolution'  in my last comment is the number of modules/networks. It means that in our analyse, we got 7 modules from the previous step. In other words, the results HMM function sent back to me, for example the 'Gamma.mat' file is a matrix with 7 columns. 

And as to transition probability matrix, we observed that each state had nearly equal probability transferring to any other state (we can not suspect whether one state was more related a certain state based on this kind of result) in K = 2, 3, 4 (the number of states). 

And now we don't know if we did something wrong, or these were due to limitation of the number of networks. I'm not sure if I depict it clearly this time.

Thank you and best wishes.

**Comment by vidaurre** [comment]
No worries. 

Can you copy your transition probability matrix here?

**Comment by lsyoath** [comment]
OK, of course.
Please wait for a minute.

**Comment by lsyoath** [comment]
This is our transition probability matrix below. When my first time trying to do HMM analyse, I just corrected on your example scripts shared here, and I just found out that it was restricted in a range of 0~0.25. So this may be the reason why I saw matrix with similar color before, am I right ?
If it is the reason, why did the range set between 0 and 0.25? Do I need to restrict it as well ?

         0    0.3771    0.3078    0.3152
    0.3178         0    0.3370    0.3452
    0.3421    0.3540         0    0.3039
    0.2491    0.3795    0.3715         0
